CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/scatter3d/Scatter3dArranger.java

```diff
@@ -0,0 +1,250 @@
+        final boolean zLogarithmic;
+        if (params != null) {
+            xDimension = params.getXDimension();
+            yDimension = params.getYDimension();
+            zDimension = params.getZDimension();
+            xLogarithmic = params.getLogarithmicX();
+            yLogarithmic = params.getLogarithmicY();
+            zLogarithmic = params.getLogarithmicZ();
+        } else {
+            return;
+        }
+
+        final int xAttr = wg.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.X.getName());
+        final int yAttr = wg.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.Y.getName());
+        final int zAttr = wg.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.Z.getName());
+
+        final int xDim = wg.getAttribute(GraphElementType.VERTEX, xDimension);
+        final int yDim = wg.getAttribute(GraphElementType.VERTEX, yDimension);
+        final int zDim = wg.getAttribute(GraphElementType.VERTEX, zDimension);
+        final int vxCount = wg.getVertexCount();
+
+        float maxX = 0;
+        float minX = 0;
+        float maxY = 0;
+        float minY = 0;
+        float maxZ = 0;
+        float minZ = 0;
+        boolean firstVals = true;
+
+        if (vxCount > 0) {
+            final BitSet vertices = GraphUtilities.vertexBits(wg);
+            int vxPos = 0;
+            final int[] vxOrder = new int[vxCount];
+            for (int vxId = vertices.nextSetBit(0); vxId >= 0; vxId = vertices.nextSetBit(vxId + 1)) {
+                if (Thread.interrupted()) {
+                    throw new InterruptedException();
+                }
+                vxOrder[vxPos++] = vxId;
+                vertices.clear(vxId);
+            }
+
+            float xVal;
+            float yVal;
+            float zVal;
+
+            for (int i = 0; i < vxPos; i++) {
+                if (Thread.interrupted()) {
+                    throw new InterruptedException();
+                }
+                final int vxId = vxOrder[i];
+
+                xVal = getFloatValueFromObject(wg.getObjectValue(xDim, vxId), xLogarithmic);
+                yVal = getFloatValueFromObject(wg.getObjectValue(yDim, vxId), yLogarithmic);
+                zVal = getFloatValueFromObject(wg.getObjectValue(zDim, vxId), zLogarithmic);
+
+                if (firstVals) {
+                    firstVals = false;
+                    maxX = xVal;
+                    minX = xVal;
+                    maxY = yVal;
+                    minY = yVal;
+                    maxZ = zVal;
+                    minZ = zVal;
+                } else {
+                    minX = Math.min(xVal, minX);
+                    maxX = Math.max(xVal, maxX);
+                    minY = Math.min(yVal, minY);
+                    maxY = Math.max(yVal, maxY);
+                    minZ = Math.min(zVal, minZ);
+                    maxZ = Math.max(zVal, maxZ);
+                }
+
+                wg.setFloatValue(xAttr, vxId, xVal);
```

Comment:
@arcturus2 - I don't think it's an issue, but could you cause issues reading and writing to attributes in a loop like this? Or are the attributes only actually written on commit?

Reply:
@cygnus-x-1 a great observation mate - but your right it will only matter on commit.